core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -318,7 +321,8 @@ private void addConnectorForListener(HttpConfiguration httpConfiguration,
       connector.setName(listener.getName());
     }
 
-    connectors.add(connector);
+    metricsListenerConnectors.add(connector);
+    rateLimitListenerConnectors.add(connector);
```

Comment:
I think this doesn't work because it is still one connector (i.e same connector in two list).

@mattwong949 another approach for this is to merge `MetricsListener` and `RateLimitNetworkTrafficListener` into one class, wdyt?